ddtrace/internal/packages.py

```diff
@@ -307,15 +311,18 @@ def _packages_distributions() -> t.Mapping[str, t.List[str]]:
     True
     """
     try:
-        import importlib.metadata as importlib_metadata
-    except ImportError:
-        import importlib_metadata  # type: ignore[no-redef]
+        try:
+            import importlib.metadata as importlib_metadata
+        except ImportError:
+            import importlib_metadata  # type: ignore[no-redef]
 
-    pkg_to_dist = collections.defaultdict(list)
-    for dist in importlib_metadata.distributions():
-        for pkg in _top_level_declared(dist) or _top_level_inferred(dist):
-            pkg_to_dist[pkg].append(dist.metadata["Name"])
-    return dict(pkg_to_dist)
+        pkg_to_dist = collections.defaultdict(list)
+        for dist in importlib_metadata.distributions():
+            for pkg in _top_level_declared(dist) or _top_level_inferred(dist):
+                pkg_to_dist[pkg].append(dist.metadata["Name"])
+        return dict(pkg_to_dist)
+    except Exception:
```

Comment:
where is the exception happening? should we try to be more targeted?

e.g. just skip specific dists if an exception occurs?

```python
for dist in importlib_metadata.distributions():
    try:
        pass
    except Exception:
        pass
return dict(pkg_to_dist)
```

?

Reply:
I just added that to be extra careful here, we don't want unchecked exception here, I suppose.